src/Hedgehog/Property.fs

```diff
@@ -4,15 +4,15 @@ open System
 
 [<Struct>]
 type Property<'a> =
-    | Property of Gen<Journal * Outcome<'a>>
+    | Property of PropertyConfig * Gen<Journal * Outcome<'a>>
```

Comment:
What about making this into a record?  How about this?
```fsharp
{ Config: PropertyConfig
  Generator: Gen<Journal * Outcome<'a>> }
```

Reply:
I'm okay with this, I think we should also mark the internals as `private`, and use lower camel case for the names:

```fsharp
type Property<'a> = {
    config: PropertyConfig
    generator: Gen<Journal * Outcome<'a>>
}
```

In any case, we should make 2 accessor functions, and only use them. They can be marked as `inline` if we have performance concerns.

```fsharp
let inline config (property : Property<'a>) : PropertyConfig =
    property.config

// This already exists as 'toGen'
let inline generator (property : Property<'a>) : PropertyConfig =
    property.generator
```

Thoughts?

Reply:
I am ok with removing the public access scope.  However, I think the access scope should be `internal` instead of private.

I think we should follow the [F# naming guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#use-pascalcase-for-type-declarations-members-and-labels) and use Pascal case for record field names.

The most common operation is mapping, not (just) getting.  As such, I think we should create a submodule inside Property like this (@dharmaturtle knows what I mean):
```fsharp
module Config =
  let get p = p.Config
  let set p c = { p with Config = c }
  let map f p = { p with Config = f p.Config }
```